crates/xilem_html/src/view_ext.rs

```diff
@@ -15,88 +17,71 @@ pub trait ViewExt<T, A>: View<T, A> + Sized {
     >(
         self,
         f: F,
-    ) -> e::OnClick<T, A, Self, F, OA>;
+    ) -> events::OnClick<T, A, Self, F, OA> {
+        events::on_click(self, f)
+    }
+
     /// Add an `ondblclick` event listener.
     fn on_dblclick<
         OA: OptionalAction<A>,
         F: Fn(&mut T, &Event<web_sys::MouseEvent, Self::Element>) -> OA,
     >(
         self,
         f: F,
-    ) -> e::OnDblClick<T, A, Self, F, OA>;
+    ) -> events::OnDblClick<T, A, Self, F, OA> {
+        events::on_dblclick(self, f)
+    }
+
     /// Add an `oninput` event listener.
     fn on_input<
         OA: OptionalAction<A>,
         F: Fn(&mut T, &Event<web_sys::InputEvent, Self::Element>) -> OA,
     >(
         self,
         f: F,
-    ) -> e::OnInput<T, A, Self, F, OA>;
+    ) -> events::OnInput<T, A, Self, F, OA> {
+        events::on_input(self, f)
+    }
+
     /// Add an `onkeydown` event listener.
     fn on_keydown<
         OA: OptionalAction<A>,
         F: Fn(&mut T, &Event<web_sys::KeyboardEvent, Self::Element>) -> OA,
     >(
         self,
         f: F,
-    ) -> e::OnKeyDown<T, A, Self, F, OA>;
+    ) -> events::OnKeyDown<T, A, Self, F, OA> {
+        events::on_keydown(self, f)
+    }
+
     fn on_blur<
         OA: OptionalAction<A>,
         F: Fn(&mut T, &Event<web_sys::FocusEvent, Self::Element>) -> OA,
     >(
         self,
         f: F,
-    ) -> e::OnBlur<T, A, Self, F, OA>;
-    /// Apply a CSS class to the child view.
-    fn class(self, class: impl Into<Cow<'static, str>>) -> Class<Self> {
-        crate::class::class(self, class)
+    ) -> events::OnBlur<T, A, Self, F, OA> {
+        events::on_blur(self, f)
     }
-}
 
-impl<T, A, V: View<T, A>> ViewExt<T, A> for V {
-    fn on_click<
-        OA: OptionalAction<A>,
-        F: Fn(&mut T, &Event<web_sys::MouseEvent, Self::Element>) -> OA,
-    >(
-        self,
-        f: F,
-    ) -> e::OnClick<T, A, Self, F, OA> {
-        e::on_click(self, f)
-    }
-    fn on_dblclick<
-        OA: OptionalAction<A>,
-        F: Fn(&mut T, &Event<web_sys::MouseEvent, Self::Element>) -> OA,
-    >(
-        self,
-        f: F,
-    ) -> e::OnDblClick<T, A, Self, F, OA> {
-        e::on_dblclick(self, f)
+    fn adapt<ParentT, ParentA, F>(self, f: F) -> Adapt<ParentT, ParentA, T, A, Self, F>
```

Comment:
I'm not entirely sure if we should use the xilem_core nodes via the builder "attribute" pattern for every HTML element.
As previously discussed on zulip, IMO at some point we want to generate (almost) all possible HTML element attributes strongly typed in a similar fashion as wasm-bindgen via "WebIDL" files.

So I think that only real HTML attributes should be used in a builder "attribute" pattern fashion.

But I think this whole topic needs to be discussed in more detail (by more people) on zulip as it's quite foundational.

Reply:
Or maybe directly in a github issue, so that it has more potential surface for external people.